File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    ''' Transcribes input dna_strand (string) to RNA.
        Outputs this as a string
    '''

    nucleotide_translations =   { 'G': 'C' 
                                , 'C': 'G'
                                , 'T': 'A'
                                , 'A': 'U' }

    
    rna_strand = ''
    
    for nucleotide in dna_strand:
        if nucleotide not in nucleotide_translations:
            raise ValueError("Invalid DNA strand input. Please check")
        rna_strand += nucleotide_translations[nucleotide]
  
    return rna_strand

File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"

namespace rna_transcription {

	using namespace std;

	const char to_rna(const char& nucleotide)
	{
		return nucleotide_translations.at(nucleotide);
	}

	const string to_rna(const string& dna_strand)
	{
		string rna_strand;

		for (const char& nucleotide : dna_strand)
			rna_strand += to_rna(nucleotide);

		return rna_strand;
	}
}  // namespace rna_transcription
